### ISSUE TITLE
Rendering of Modded Armor/Clothing Broken

### ISSUE THREAD

**Issue by SalmonSays** [Tommsy64/SmartMovingReloaded#5]
I'm gunna make a separate issue for this if that's alright with you @Tommsy64 -

So here are visual samples of what goes wrong when using certain modded armors alongside SMR:

https://i.imgur.com/xxrKtlY.jpg
-The body fits more or less naturally when facing south. In these examples I'll be using Elementium Armor from Botania but it'll happen on a lot of modded armors like those from Construct's Armory, Jetpacks from SimplyJetpacks, Mekanism armors, etc. The only armors that I've seen work so far are ones implemented in a very "vanilla" way like TechReborn.

https://i.imgur.com/CXJsx0S.jpg
-When turning, the body maintains idle animations and action motions like when clicking/mining, but doesn't match the movement of the player rotationally.

https://i.imgur.com/YdldmHa.jpg
-When crouching in the correct direction (South), the armor model crouches but clips through the body and is positioned incorrectly.

https://i.imgur.com/eyBhmiN.jpg
-When crawling the armor stays in the crouched position.

https://i.imgur.com/IFQwHz4.jpg
-And continues to not inherit rotation properly.

This has historically been a problem with the Render Player API used in previous versions of Smart Moving because of unexpected ways certain mods access RenderPlayer, ModelPlayer, and ModelBiped (cct, bqj, and bpx respectively). Lifted straight from the readme, Render Player API Enhancer for 1.12.x has attempted to fix this implementation by violently forcing all classes that:

* extend 'net.minecraft.client.model.ModelBiped' and
* have a canonical name that matches at least one of the candidate class names or patterns

to use Render Player API by

* changing their super class from 'net.minecraft.client.model.ModelBiped' to 'api.player.model.ModelPlayerArmor' and
* removing all locally declared methods that overwrite methods used by Render Player API in case they dont't call their super method at least on one execution path.

I'm hopeful there's a more tasteful way of fixing this issue than forcing classes like that, but I have yet to discover it- though admittedly I'm still wrapping my head around using mixins properly.

**Comment by Tommsy64** [comment]
I believe using a transformer to change the superclass of classes that extend `net.minecraft.client.model.ModelBiped` (just like RenderPlayerAPI did) will be the best way to handle this.

SpongePowered does exactly this:
* [SpongeSuperclassRegistry](https://github.com/SpongePowered/SpongeCommon/blob/bleeding/src/main/java/org/spongepowered/common/launch/transformer/SpongeSuperclassRegistry.java)
* [SpongeSuperclassTransformer ](https://github.com/SpongePowered/SpongeCommon/blob/bleeding/src/main/java/org/spongepowered/common/launch/transformer/SpongeSuperclassTransformer.java)
* [SpongeLaunch](https://github.com/SpongePowered/SpongeCommon/blob/bleeding/src/main/java/org/spongepowered/common/launch/SpongeLaunch.java#L142-L153)

I tried looking for a few minutes but couldn't find where the TechReborn armors are implemented.

**Comment by Tommsy64** [comment]
Running SmartMoving (the port) 1.12 with RenderPlayerAPIEnhancer and Botania makes two sets of armor appear, one that is correct and one that remains facing only one direction.

![2018-09-03_14 59 26](https://user-images.githubusercontent.com/3900951/44999309-47d09600-af8a-11e8-8b90-84f5055512df.png)

**Comment by SalmonSays** [comment]
@Tommsy64 Quick Update- I was poking around the repository for Resize Potion, whose source is available here:

https://github.com/CammiePone/Resize-Potion

I noticed that their solution for handling armor plays **really nicely** with Botania/custom armor models for scaling, rotation, translation, etc.

![2018-11-30_17 23 08](https://user-images.githubusercontent.com/11697389/49320013-20c02c80-f4c5-11e8-8923-ae9a3ecf4154.png)

It's my hope these methods can be retrofitted to SmartMovingReloaded, as I'm super confident this is enough of a fix/update to warrant a totally usable beta for end-users.

**Comment by Tommsy64** [comment]
@SalmonSays I appreciate the help! Unfortunately, the Resize-Potion mod simply [changes](https://github.com/CammiePone/Resize-Potion/blob/a652bff40b1f02c410c8949a9db7e76a5a50f82c/src/main/java/com/camellias/resizer/potions/handler/PotionHandler.java#L150) a "size" value built into all entities by Vanilla. (I believe this is used to control the size of slime mobs). This value acts as a simple multiplier for the *entire* rendering of the entity.

To add all the custom movement animations, SmartMoving completely replaces the rendering of the player entity. As other mods create their custom armor for the Vanilla player renderer, there is an inherent incompatibility.

Fortunately, [MoBends](https://github.com/iwoplaza/MoBends) appears to have a way of [modifying the vanilla renderer](https://github.com/iwoplaza/MoBends/blob/master-1.12.2/src/main/java/net/gobbob/mobends/client/mutators/BipedMutator.java) to support custom animations rather than replacing it. Once I understand how it is done, I'll see if I can do something similar.
